pkg/kubectl/resource/builder.go

```diff
@@ -240,6 +240,13 @@ func (b *Builder) RequireNamespace() *Builder {
 	return b
 }
 
+// NameParam accepts the name for a resource. Meant to be used in combination with a default
+// resource type set by ResourceTypes.
+func (b *Builder) NameParam(name string) *Builder {
+	b.names = append(b.names, name)
```

Comment:
I thought `names` were specifically just the names, not name OR resource/name.  I'd really like to be able to say, "Here's an arg, I don't know if it's a name or a resource/name, but if it's just a name use 'pods' as a default resource.  Otherwise, use the resource from the arg".